### PR TITLE
Converted into a loop

### PR DESCRIPTION
**Pull request by nicholasmullikin**

### CODE REVIEW

File: signal-compiler.c [hunk 1]
```diff
@@ -38,53 +38,54 @@ int main(int ac, char *av[]) {
...
-    }
+      // - write(pptr, pgm_buf, PGM_H * PGM_W)
+      // - write(pptr, pgm_buf[ii], PGM_H)
+      // - write(pptr, &pgm_buf[ii][jj], 1)   <-- use this one (byte-order reasons)
+      for (unsigned jj = 0; jj < PGM_H; jj++) {
+          for (unsigned ii = 0; ii < PGM_W; ii++) {
+              write(pptr, &pgm_buf[ii][jj], 1);
+          }
+      }
+      printf("wrote %ui to pgm\n", sizeof(pgm_buf));
```
Comment (nicholasmullikin) [inline signal-compiler.c:87]: Also I changed this because the make wouldn't let it pass.

**Comment by jleightcap** [comment]
not a huge fan of wrapping this in a loop, that's what [this](https://github.com/accessibility-oscilloscope/buildroot/blob/master/accessibility-oscilloscope/overlay/usr/bin/fifo-wrapper) was for. a main loop *is* symmetric with other things, but makes testing this in isolation way harder.

File: signal-compiler.c (hunk 1 above)
Comment (jleightcap) [reply signal-compiler.c:87]: oh weird! i think `%ld` was what `clang` gave a suggestion for. good to know

**Comment by nicholasmullikin** [comment]
> not a huge fan of wrapping this in a loop, that's what [this](https://github.com/accessibility-oscilloscope/buildroot/blob/master/accessibility-oscilloscope/overlay/usr/bin/fifo-wrapper) was for. a main loop *is* symmetric with other things, but makes testing this in isolation way harder.

Ok but we're gonna have to move the mkfifo part of it or remove it from the init script

**Comment by jleightcap** [comment]
ah, right! the `fifo-wrapper` was before we had the 'distinct making fifos' step in init. i'll open an issue in buildroot.

**Comment by jleightcap** [comment]
see [issue](https://github.com/accessibility-oscilloscope/buildroot/issues/16).

let me open a PR there and i'll add you as a reviewer.

**Comment by nicholasmullikin** [comment]
Wait do you get that error in the print statement if you clone and make or is that just mine

**Comment by jleightcap** [comment]
> Wait do you get that error in the print statement if you clone and make or is that just mine

no, it's built just fine for me locally and on buildroot. i'm assuming you're building locally on raspbian? i believe `%ld` should work:

grim-2022-03-23-164403

what was the specific error you were getting?

**Comment by jleightcap** [comment]
> see [issue](https://github.com/accessibility-oscilloscope/buildroot/issues/16).
> 
> let me open a PR there and i'll add you as a reviewer.

[PR](https://github.com/accessibility-oscilloscope/buildroot/pull/17)

**Comment by nicholasmullikin** [comment]
This was the issue. Not really worth investigating if its Raspbian specific.

```
pi@pi:~/accessibility-oscilloscope/signal-compiler $ make
cc signal-compiler.c -Wall -Wextra -Werror -pedantic -std=c99 -lm -o signal-compiler
signal-compiler.c: In function ‘main’:
signal-compiler.c:86:19: error: format ‘%ld’ expects argument of type ‘long int’, but argument 2 has type ‘unsigned int’ [-Werror=format=]
   86 |   printf("wrote %ld to pgm\n", sizeof(pgm_buf));
      |                 ~~^            ~~~~~~~~~~~~~~~
      |                   |            |
      |                   long int     unsigned int
      |                 %d
cc1: all warnings being treated as errors
make: *** [Makefile:8: signal-compiler] Error 1

```

**Comment by jleightcap** [comment]
> This was the issue. Not really worth investigating if its Raspbian specific.
> 
> ```
> pi@pi:~/accessibility-oscilloscope/signal-compiler $ make
> cc signal-compiler.c -Wall -Wextra -Werror -pedantic -std=c99 -lm -o signal-compiler
> signal-compiler.c: In function ‘main’:
> signal-compiler.c:86:19: error: format ‘%ld’ expects argument of type ‘long int’, but argument 2 has type ‘unsigned int’ [-Werror=format=]
>    86 |   printf("wrote %ld to pgm\n", sizeof(pgm_buf));
>       |                 ~~^            ~~~~~~~~~~~~~~~
>       |                   |            |
>       |                   long int     unsigned int
>       |                 %d
> cc1: all warnings being treated as errors
> make: *** [Makefile:8: signal-compiler] Error 1
> ```

thanks for posting full output. [good explanation on stack overflow](https://stackoverflow.com/a/5943869), since we're using c99 anyways `%zu` seems to be the actually portable way of doing this.

if you want to verify, added fix to #2

**Pull request closed without merging**